src/utilities/time.ts

```diff
@@ -66,6 +164,149 @@ function parseDurationString(durationString: string): ParsedDurationString | nev
   throw Error('Must be of format: 1y 3d 2h 24m 35s 18ms 70us');
 }
 
+/**
+ * Format a duration object to a day of year string.
+ *
+ * @example
+ * formatDurationToDoy({
+ *   years: 0,
+ *   days: 1,
+ *   hours: 0,
+ *   minutes: 45,
+ *   seconds: 0,
+ *   milliseconds: 10,
+ *   microseconds: 0,
+ * })
+ *
+ * result: '1970-1T00:45:00.010'
+ *
+ * @param {ParsedDurationString} duration - The duration object to format.
+ * @returns {string} - The formatted day of year string.
+ */
+function formatDurationToDoy(duration: ParsedDurationString): string {
```

Comment:
can this function name follow the "convert" naming convention in this file?

Reply:
renamed